src/NServiceBus.Core/Routing/RoutingMappingSettings.cs

```diff
@@ -32,13 +31,13 @@ internal RoutingMappingSettings(SettingsHolder settings)
         public Publishers Publishers => GetOrCreate<Publishers>();
 
         /// <summary>
-        /// Sets a distribution strategy for a given subset of message types.
+        /// Sets a distribution strategy for a given endpoint.
         /// </summary>
+        /// <param name="endpointName">The name of the logical endpoint the given strategy should apply to.</param>
         /// <param name="distributionStrategy">The instance of a distribution strategy.</param>
-        /// <param name="typeMatchingRule">A predicate for determining the set of types.</param>
-        public void SetMessageDistributionStrategy(DistributionStrategy distributionStrategy, Func<Type, bool> typeMatchingRule)
+        public void SetMessageDistributionStrategy(string endpointName, DistributionStrategy distributionStrategy)
```

Comment:
With this API, you'd have to call it for every endpoint to apply a custom strategy. Would it be worth it to add a way to change the default strategy in the policy, so that you could replace it once for all endpoints?

Or maybe having different strategies within a single policy is an advanced scenario, and instead we should let the policy be swapped out?

Ultimately, having a Policy that contains multiple different Strategies seems like the advanced case, and a Policy that applies a single custom Strategy seems more the more common use case.